javascripts/discourse/initializers/init-topic-excerpts.js

```diff
@@ -15,25 +15,16 @@ export default {
 
   initialize() {
     withPluginApi("1.34.0", (api) => {
-      const router = api.container.lookup("service:router");
+      const discovery = api.container.lookup("service:discovery");
       api.registerValueTransformer(
         "topic-list-item-expand-pinned",
         ({ value, context }) => {
           const overrideEverywhere =
             enabledCategories.length === 0 && enabledTags.length === 0;
           const overrideInCategory = enabledCategories.includes(
-            excerptsViewingCategoryId(
-              router.currentRouteName,
-              router.currentRoute.attributes.category.id
-            )
-          );
-          const overrideInTag = enabledTags.includes(
-            excerptsViewingTag(
-              router.currentRouteName,
-              router.currentRoute.attributes.id,
-              router.currentRoute.attributes.tag.id
-            )
+            discovery.category
```

Comment:
enabledCategories is a list of IDs, right?

```suggestion
            discovery.category?.id
```

Reply:
🙃 , I will just bite the bullet and write the tests here...